kbs/src/attestation/keylime.rs

```diff
@@ -0,0 +1,428 @@
+// Copyright (c) 2025 Red Hat.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::attestation::backend::{Attest, IndependentEvidence};
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use base64::{
+    prelude::{BASE64_STANDARD, BASE64_URL_SAFE_NO_PAD},
+    Engine,
+};
+use derivative::Derivative;
+use ear::{
+    Algorithm, Appraisal, Bytes, Ear, Extensions, Nonce, RawValue, RawValueKind, VerifierID,
+};
+use jsonwebtoken::jwk;
+use kbs_types::{RuntimeData, Tee, TeePubKey};
+use openssl::{pkey::Public, rsa::Rsa, x509::X509};
+use reqwest::{
+    header::CONTENT_TYPE,
+    tls::{Certificate, Identity},
+    Client, ClientBuilder,
+};
+use serde::Deserialize;
+use serde_json::{json, Value};
+use sev::firmware::guest::AttestationReport as SnpReport;
+use std::{collections::BTreeMap, fs};
+use time::{Duration, OffsetDateTime};
+
+#[derive(Clone, Debug, Derivative, Deserialize, PartialEq, Default)]
+pub struct KeylimeTeeConfig {
+    pub base_url: String,
+    pub api_version_major: u8,
+    pub api_version_minor: u8,
+    pub developer_name: String,
+    pub build_name: String,
+    pub cv_ca_path: String,
+}
+
+pub struct KeylimeTeeHandler {
+    config: KeylimeTeeConfig,
+    client: Client,
+    cert: X509,
+    priv_key_pem: Vec<u8>,
+    pub_key: Rsa<Public>,
```

Comment:
These keys are all used for making the JWT attestation token, right? Do we do anything to validate the response coming from keylime (besides https)? I guess keylime itself doesn't return any kind of JWT.

Reply:
Ok, I guess it could be more clear which is which.